otter/util/fp.py

```diff
@@ -78,3 +79,10 @@ def predicate_any(*preds):
     with or operator
     """
     return lambda *a, **kw: any(p(*a, **kw) for p in preds)
+
+
+def obj_assoc(o, **k):
```

Comment:
so, I was going to say that I named it after the function in the proposed `objtoolz` package (to be added to toolz), except I apparently got the name backwards:

https://github.com/pytoolz/toolz/pull/195

they use `assoc_obj` instead of `obj_assoc`. I'd prefer to stick with their names so we can switch to that library once it's included and released.